vmm/src/api/openapi/cloud-hypervisor.yaml

```diff
@@ -499,21 +499,28 @@ components:
           type: string
       description: Information about a PCI device
 
+    PayloadConfig:
+      type: object
+      properties:
+        kernel:
+          type: string
+        cmdline:
+          type: string
+        initramfs:
+          type: string
+      description: Payloads to boot in guest
+
     VmConfig:
       required:
-        - kernel
+        - payload
```

Comment:
If it's required, then why is it an `Option<PayloadConfig>` in the `VmConfig` in `vmm/src/config.rs`? Is it because of the backward compatibility? And does that mean you're planning on moving from `Option<PayloadConfig>` to `PayloadConfig` when we completely remove the old support (in 2 releases time)?

Reply:
The `Option<PayloadConfig>` is there to support the handling of the "API only mode" where we just start the VMM without booting a payload. In the HTTP REST API the payload is required but when parsing the command line we might not have any payload parameters. Currently the detection of the "API only mode" has manual parameter checks and the use of the `Option` type is there to allow this to be simplified.

Reply:
Ah yes I forgot about the "API only mode", thanks for the clarification.